util/conditions/patch_test.go

```diff
@@ -27,6 +27,106 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+func TestNewPatchOperations(t *testing.T) {
+	fooTrue := TrueCondition("foo")
+	fooFalse := FalseCondition("foo", "reason foo", clusterv1.ConditionSeverityInfo, "message foo")
+
+	tests := []struct {
+		name    string
+		before  Getter
+		after   Getter
+		want    Patch
+		wantErr bool
+	}{
+		{
+			name:    "nil before return error",
+			before:  nil,
+			after:   getterWithConditions(),
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name:   "nil after return error",
+			before: getterWithConditions(),
+			after:  nil,
+			want:   nil,
+		},
+		{
+			name:   "nil Interface after return error",
```

Comment:
I'm not sure I understand what we are trying to do, but this case does not return an error

Reply:
Probably related to my comment above. nilGetter returns a Cluster where conditions are nil, not some "other" kind of nil :)